scheduler/reconciler/reconcile_node.go

```diff
@@ -69,13 +69,17 @@ func (nr *NodeReconciler) Compute(
 	compatHadExistingDeployment := nr.DeploymentCurrent != nil
 
 	result := new(NodeReconcileResult)
-	deploymentComplete := true
+	var deploymentComplete bool
 	for nodeID, allocs := range nodeAllocs {
 		diff, deploymentCompleteForNode := nr.computeForNode(job, nodeID, eligibleNodes,
 			notReadyNodes, taintedNodes, canaryNodes[nodeID], canariesPerTG, required,
 			allocs, terminal, serverSupportsDisconnectedClients)
-		deploymentComplete = deploymentComplete && deploymentCompleteForNode
 		result.Append(diff)
+
+		deploymentComplete = deploymentCompleteForNode
+		if deploymentComplete {
+			break
+		}
```

Comment:
I don't understand this change. Wouldn't this complete the deployment on the first completed node?

Reply:
No. Unlike in the cluster reconciler, the `isDeploymentComplete` method in the node reconciler takes into account the total desired count, so it will only ever mark the deployment as complete if there's at least as many healthy allocs as the desired total or desired canary.